internal/dataplane/kong_client.go

```diff
@@ -628,7 +640,10 @@ func extractBrokenObjectsFromUpdateError(err error) ([]fallback.ObjectHash, erro
 // sendOutToGatewayClients will generate deck content (config) from the provided kong state
 // and send it out to each of the configured gateway clients.
 func (c *KongClient) sendOutToGatewayClients(
-	ctx context.Context, s *kongstate.KongState, config sendconfig.Config,
+	ctx context.Context,
+	s *kongstate.KongState,
+	config sendconfig.Config,
+	isFallback bool,
```

Comment:
It's a little hard to follow with such an API at least for me. Maybe having two methods
- `sendOutToGatewayClients(ctx context.Context,	s *kongstate.KongState,	config sendconfig.Config)`
- `sendOutToGatewayClientsFallback(ctx context.Context,	s *kongstate.KongState,	config sendconfig.Config)`

and inside calling the above with `isFallback = false/true`. WDYT?

Reply:
Yeah, I don't like this API either, but we have to pass this flag up from `KongClient.Update` and `tryRecoveringWithFallbackConfiguration` down to `sendToClient` through `sendOutToGatewayClients` and `maybeSendOutToKonnectClient`. If we'd like to have a dedicated method for each, we'd end up with:

- `sendToClientFallback`
- `sendToClientRegular`
- `sendToClient` (implementation with fallback flag param)
- `sendOutToGatewaysClientsFallback`
- `sendOutToGatewaysClientsRegular`
- `sendOutToGatewaysClients` (implementation with fallback flag param)
- `maybeSendOutToKonnectClientFallback`
- maybeSendOutToKonnectClientRegular`
- `maybeSendOutToKonnectClient` (implementation with fallback flag param)

which seems to me like an overkill.

I considered below options:
- passing the fallback flag in the context (it would be quite implicit and easy to miss)
- extending `KongState` with `Meta` struct that would contain the `IsFallback` flag (it breaks the `KongState` - currently it only contains Kong entities and I think it's better to leave it this way)
- passing `ConfigPushMetadata` with the `IsFallback` field instead of the raw flag (this seemed to me like overkill, but maybe this would be cleaner as we could have some godocs explaining the meaning of it). WDYT?

Reply:
I see the cardinality rise. Let's leave it for now as it is for now